src/projects/providers/rtspc/rtspc_stream.cpp

```diff
@@ -914,6 +1059,10 @@ namespace pvd
 			}
 			else
 			{
+				if(time(nullptr) - _last_check_time >= 30){
```

Comment:
I think the timeout should be taken from the SETUP response of the RTSP camera

```
[04-27 19:21:56.360] I [StreamCollector:630581] RtspcStream | rtspc_stream.cpp:448 | Response SETUP: Response: RTSP/1.0 200 OK
Connection: Keep-Alive
CSeq: 11
Date: Thu, 27 Apr 2023 10:21:45 GMT
Session: 2613658800411225924828208291162;timeout=120
Transport: RTP/AVP/TCP;unicast;interleaved=20-21;ssrc=78CD202E
```

Reply:
I agree. It is recommended to update at about half of the SETUP's timeout.

Reply:
I have a camera that shown timeout=0 and drop connection after 90 sec

Reply:
Then maybe use the timeout if provided and fall back to 30s if there is no timeout set or it is 0?